### ISSUE TITLE
Token changes?

### ISSUE THREAD

**Issue by vespino** [phonegap-build/PushPlugin#520]
I just read the iOS token changes with every update. Is that correct? How do you guys cope with this issue because I guess the old token can't be used anymore and unless the user opens the app, a new token isn't send to the server.

Help? :)

**Comment by Hirbod** [comment]
There is no chance for you. You need to wait, till the app is reopened.
I've managed to make the "register" call on every app start, and associated the token with the user-id on mysql, and when changed, I replaced, if not -> go on

**Comment by vespino** [comment]
So what happens when a user doesn't use the app for a long time? No push notifications?

**Comment by Hirbod** [comment]
Right. Push is nothing you can count or rely on. It's just FF - fire and forget

**Comment by vespino** [comment]
Very strange way of working of you ask me. Is this just iOS or Android to?

**Comment by Hirbod** [comment]
Both. It is what it is by design. This is why you should never do business relevant stuff on behalf of push. You should read some articles about that. It is intended to be a service, a nice to have, but always remember: there is no way ever to determine your push was arrived on the phone. The user could have deactivated it, or servers were too busy, phone was shut down and max delivery time exceeded and so on...

**Comment by Hirbod** [comment]
http://stackoverflow.com/questions/25830597/how-to-know-push-notification-delivery-status

**Comment by Hirbod** [comment]
You have the option to use apples feedback service, android has some options too. But this will just give you a list of tokens where push could not be send too

**Comment by vespino** [comment]
So I guess it's up to my client to keep people using the app.

Thanks for you answers.
